storybook/src/Storybook/PagesList.qml

```diff
@@ -1,22 +1,89 @@
 import QtQuick 2.14
 import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import Qt.labs.folderlistmodel 2.14
 
 ListView {
     id: root
 
     spacing: 5
     clip: true
 
-    property string currentPage
-    signal pageSelected(string page)
+    property url currentPage
+    property string filter
+
+    ScrollBar.vertical: ScrollBar {}
+
+    Component {
+        id: pageDelegate
+        ItemDelegate {
+            width: parent.width
+
+            text: model_fileBaseName.replace(/Page+$/, "") // strip trailing "Page" string
+            highlighted: root.currentPage === model_fileUrl
+            font.bold: highlighted
+
+            onClicked: root.currentPage = model_fileUrl
+        }
+    }
+
+    Component {
+        id: folderDelegate
 
-    delegate: Button {
-        width: parent.width
+        ColumnLayout {
+            id: categoryDelegate
+            width: parent.width
+            spacing: 5
+
+            property bool expanded: true
+            readonly property string folder: model_fileUrl
+
+            ItemDelegate {
+                Layout.fillWidth: true
+                text: "%1 %2 (%3)".arg(categoryDelegate.expanded ? "▼" : "▶").arg(model_fileName).arg(pageRepeater.count)
+                onClicked: categoryDelegate.expanded = !categoryDelegate.expanded
+                highlighted: root.currentPage.toString().startsWith(categoryDelegate.folder)
+                font.bold: highlighted
+            }
+
+            ColumnLayout {
+                Layout.fillWidth: true
+                Layout.leftMargin: 16
+                visible: categoryDelegate.expanded
+                Repeater {
+                    id: pageRepeater
+                    Layout.fillWidth: true
+                    model: FolderListModel {
+                        folder: categoryDelegate.folder
+                        caseSensitive: false
+                        showDirs: false
+                        showFiles: true
+                        nameFilters: [`*${root.filter}*Page.qml`]
+                        showDotAndDotDot: false
+                        sortField: FolderListModel.Name
+                    }
+
+                    delegate: ItemDelegate {
```

Comment:
Something like this would create a tree like view and we could support nested folders for the qml files:
```
model: FolderListModel {
                         folder: categoryDelegate.folder
                         caseSensitive: false
                         showDirs: true
                         showFiles: true
                         nameFilters: [`*${root.filter}*Page.qml`]
                         showDotAndDotDot: false
                         sortField: FolderListModel.Name
                     }
                     delegate: rowDelegate
                }
            }
        }
    }

    delegate: rowDelegate

    Component {
        id: rowDelegate
        Loader {
                property url model_fileUrl: model.fileURL
                property string model_fileName: model.fileName
                property string model_fileBaseName: model.fileBaseName

                sourceComponent: model.fileIsDir ? folderDelegate : pageDelegate
                width: parent.width
        }    
    }
}
```

Reply:
Cool, gonna give it a try on Monday

Reply:
@alexjba done, works perfectly (with some minor adjustments :)

![image](https://user-images.githubusercontent.com/5377645/201390301-9b9510b6-c5b9-4ef9-b7b7-98d11fa04de5.png)